nrfx/mdk/system_nrf52840.c

```diff
@@ -130,7 +130,21 @@ void SystemInit(void)
             NRF_POWER->RESETREAS =  ~POWER_RESETREAS_RESETPIN_Msk;
         }
     }
-    
+
+    /* Workaround for Errata 197 "DCDC of REG0 not functional"
+     * found at the Errata document for your device located at
+     * https://infocenter.nordicsemi.com/index.jsp
+     */
+    if (nrf52_errata_197()){
+        if (NRF_POWER->DCDCEN0 == 0x1ul)
+        {
+            /* Disable REG0 */
+            NRF_POWER->DCDCEN0 = 0;
+            /* prevent REG0 stage to go to ULP mode to prevent random resets. */
+            *(volatile uint32_t *)0x40000638 = 0x01ul;
```

Comment:
This does not make much sense. The workaround for anomaly 197 is either to not use DCDC on REG0 or to prevent REG0 stage to go to ULP mode (by writing 1 to register 0x40000638). One of the options, not both together.

Reply:
In my testing, to solve the reset issue **and** draw the least amount of power, turn off the DCDCEN0 and prevent ULP.
If you just prevent ULP mode, much more current is consumed so why not shut it off as well?  Seemed like a universal solution.

Reply:
My point is that doing `*(volatile uint32_t *)0x40000638 = 0x01ul;` is just pointless when `DCDCEN0` is `0`.

Reply:
Actually, during my testing, the reset issue still happened if I only disabled `DCDCEN0`.  Both steps are required if `DCDCEN0` was ever set to `1` in order to fix the potential for random resets and to draw the least amount of power..

Reply:
Okay. That's a bit surprising. But I haven't tested it myself, I just based on what was written in the errata. Perhaps "Do not enable DCDC" is actually supposed to mean do not enable it at all after reset, and I just misunderstood this.

Reply:
> Perhaps "Do not enable DCDC" is actually supposed to mean do not enable it at all after reset, and I just misunderstood this.

Agree, and after my testing, that is what I took it to mean.